### DISCUSSION TITLE
WebView in C# .net MAUI

### DISCUSSION THREAD

**Discussion by Smanueeel** [dotnet/maui#21447] [Q&A]
Hi everyone, I'm Manuel and I'm new on GitHub, so I'm sorry for any mistake during the creation of this discussion.
I work as IT Technician and my boss assigned me to create the first app of our company. This app must be based on our web services, so I used some WebView to get our websites sections into the app. In one of these, users should compile a form and then download a document which reports the corresponding data from the form. Everything works fine, except for the fact that the download doesn't start. In fact, when I try to press the download button, nothing happens and I don't know why.

Into android manifest I set external memory writing and reading permission. Then, I defined WebView into a C# method (so it isn't defined into xml main page).

Could anyone help me?
Thank you

**Comment by ronanj2** [comment]
Have you code you could share?

**Comment by Smanueeel** [comment]
```
private async void SetServicesView(object sender, EventArgs e, string type){
    WebView webViewerS = new WebView();
    
     if (type == "frisona") {
        MainGrid.Children.RemoveAt(1);
        MainGrid.Add(ALoader);
        ALoader.IsVisible = true;
        ALoader.IsRunning = true;

        webViewerS.Source = "my/URL/source";
        Grid.SetColumn(webViewerS, 0);
        Grid.SetColumnSpan(webViewerS, 5);
        Grid.SetRow(webViewerS, 1);
        
        ALoader.IsVisible = false;
        ALoader.IsRunning = false;
        MainGrid.Children.RemoveAt(1);
        MainGrid.Add(webViewerS);
    }
```

That's my C# code. Into .xaml there's only the definition of the screen "header" and ALoader, which simply is an ActivityIndicator which I make visible while other widgets are loading.
As I said before, in "my/URL/source" there is a form. When user compile that form, they have the possibility to download a PDF (or choose other formats) pressing a button **into** the website. The form is displayed correctly within my app and all works well. The only problem is that download which doesn't start (as if I had never pressed the button).
